### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Dlewis89**
# Description

Implemented Card preview loader for discover details view.

## Checklist

- [x] Screenshots added for any UX changes
- [x] Demos attached for animations/interactions
- [x] Pointed at proper branch (`develop` not `main`)
- [x] Assigned PR to yourself
- [x] Requested review from code owners
- [x] Relevant labels added (`feature`, `documentation`, etc.)
- [x] `Closes #<issue-number>` added if this resolves a GitHub issue
- [x] Branch is up to date with develop, ran `git rebase develop` if necessary
- [x] All GitHub Actions are passing
- [x] `npm run todo-ci` ran if any todo comments were created

### CODE REVIEW

File: src/screens/loaders/CardLoader.tsx
Comment (Thenlie) [inline src/screens/loaders/CardLoader.tsx:1]: Can we rename this to `CardGalleryLoader`?

File: src/screens/loaders/CardLoader.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { ShowCardLoader, ShowListCardLoader } from '../../components';
+import useWindowSize from '../../hooks/useWindowSize';
+
+interface CardLoaderProps {
+    query?: string | null;
```
Comment (Thenlie) [inline src/screens/loaders/CardLoader.tsx:6]: Is `query` used anywhere here? If not, can we remove it?

File: src/screens/discover/DiscoverDetailScreen.tsx [hunk 2]
```diff
@@ -104,26 +105,27 @@ const DiscoverDetailScreen: React.FC = () => {
...
-                setViewState={setViewState}
-                data={data}
-                setData={setData}
-                setHash={setHash}
-                cards={cards}
-                disableAlphabeticOrderFilter={true}
-                disableResultTypeFilter={true}
-            />
+            {loading ? (
+                <CardLoader query={title} viewState={viewState} />
```
Comment (Thenlie) [inline src/screens/discover/DiscoverDetailScreen.tsx:115]: This looks really good! I think the only "problem" I see is that the filter bar is not in the loader. This not a deal breaker, but it does cause the UI to jump a bit between the loader and the content. If it is not too much to add, could you add the filter to the loader screen? `SearchResultLoader` has a good example of this. You may need to create a wrapper, something like `DiscoverDetailLoader`.

**Review by Thenlie — CHANGES_REQUESTED** [review]
In general, this looks awesome! Really appreciate all the work here, this definitely improves the UI when going to this screen.

**Comment by Dlewis89** [comment]
Hey boss! No problem, i can take a look at this tomorrow when i can and get everything squared away.

File: src/screens/loaders/CardLoader.tsx
Comment (Dlewis89) [reply src/screens/loaders/CardLoader.tsx:1]: Rename of component is good to go.

File: src/screens/loaders/CardLoader.tsx (hunk 1 above)
Comment (Dlewis89) [reply src/screens/loaders/CardLoader.tsx]: remove the unnecessary variable.

File: src/screens/discover/DiscoverDetailScreen.tsx (hunk 2 above)
Comment (Dlewis89) [reply src/screens/discover/DiscoverDetailScreen.tsx]: Screenshot 2024-11-30 at 17-34-55 Streamability
Screenshot 2024-11-30 at 17-34-19 Streamability

Should be good to go here as well. Provided updated images for the layout. Hope you had a great thanksgiving!

**Comment by Dlewis89** [comment]
Everything should be good to go. Also providing updated images here. 
Screenshot 2024-11-30 at 17-34-55 Streamability
Screenshot 2024-11-30 at 17-34-19 Streamability

**Review by Thenlie — APPROVED** [review]

**Comment by Thenlie** [comment]
The `todo-ci` tasks was missed. Going to merge anyways and fix that in a follow-up PR.

**Pull request merged**